### PR TITLE
Docs: Storage documentation

### PR DESCRIPTION
**Pull request by scholzj**
### Type of change

- Documentation

### Description

This PR modularizes the Storage documentation. It contains only assembly and reference, since the storage cannot be changed - therefore there is no procedure.

### Checklist

- [x] Update documentation

### CODE REVIEW

File: documentation/book/ref-storage.adoc [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+  zookeeper:
+    # ...
+    storage:
+      type: ephemeral
+    # ...
+----
+
+== Persistent storage
+
+Persistent storage is using K8sPersistentVolumeClaims to provisine persistent volumes for storing data.
```
Comment (ppatierno) [inline documentation/book/ref-storage.adoc]: Is K8sPersistentVolumeClaims meant to be and attribute so brackets are missing?

File: documentation/book/assembly-storage.adoc [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+// See also the complementary step on the last line of this file.
+:parent-context: {context}
+
+[id='assembly-storage-{context}']
+= Kafka and Zookeeper storage
+
+Kafka brokers and Zookeeper are stateful applications.
+They need to store data on disks.
+{ProductName} allows users to configure the type of storage which they want to use for Kafka and Zookeeper.
+Storage configuration is mandatory and has to be specified in every `Kafka` resource.
```
Comment (dmuntima) [inline documentation/book/assembly-storage.adoc]: add comma before which (storage, which they) and second person usage:
{ProductName} allows you to configure the type of storage for Kafka and Zookeeper.

File: documentation/book/ref-storage.adoc [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+  zookeeper:
+    # ...
+    storage:
+      type: ephemeral
+    # ...
+----
+
+== Persistent storage
+
+Persistent storage is using {K8sPersistentVolumeClaims} to provisine persistent volumes for storing data.
```
Comment (dmuntima) [inline documentation/book/ref-storage.adoc]: - Change  "Persistent storage uses {K8sPersistentVolumeClaims} to provisine persistent" to "Persistent storage uses {K8sPersistentVolumeClaims} to provision persistent " (typo and reduced a word)
- delete many from "provision volumes of many different types"

File: documentation/book/ref-storage.adoc [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+Persistent storage supports additional configuration options:
+
+`size` (required)::
+Defines the size of the persistent volume claim, for example, "1000Gi".
+
+`class` (optional)::
+The {ProductPlatformName} {K8SStorageClass} to use for dynamic volume provisioning.
+
+`selector` (optional)::
+Allows to select a specific persistent volumes to use.
```
Comment (dmuntima) [inline documentation/book/ref-storage.adoc]: correct article-noun agreements: "Allows to select a specific persistent volumes to use." to "Allows selecting a specific persistent volume to use."  or "Allows you to select specific persistent volumes to use."

File: documentation/book/ref-storage.adoc [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+----
+# ...
+storage:
+  type: persistent-claim
+  size: 1000Gi
+# ...
+----
+
+The following example demonstrates use of a storage class.
+
```
Comment (dmuntima) [inline documentation/book/ref-storage.adoc]: Add "the before "use".

File: documentation/book/ref-storage.adoc [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+----
+# ...
+storage:
+  type: persistent-claim
+  size: 1Gi
+  class: my-storage-class
+# ...
+----
+
+Finally, a `selector` can be used in order to select a specific labelled persistent volume which provides some needed features (such as an SSD).
```
Comment (dmuntima) [inline documentation/book/ref-storage.adoc]: change "some needed features (such as an SSD)." to "some needed features like an SSD."

File: documentation/book/ref-storage.adoc [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+    matchLabels:
+      "hdd-type": "ssd"
+  deleteClaim: true
+# ...
+----
+
+=== Persistent Volume Claim naming
+
+When the persistent storage is used, it will create Persistent Volume Claims with following names:
+
```
Comment (dmuntima) [inline documentation/book/ref-storage.adoc]: The replaceable text has to be styled in monospace and surrounded by angle brackets. http://strimzi.io/docs/master/#document-conventions-str. Change it to "`data-_<cluster-name>_-kafka-[idx]` `data-_<cluster-name>_-zookeeper-[idx]"

File: documentation/book/ref-storage.adoc [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+`data-[cluster-name]-kafka-[idx]`::
+Persistent Volume Claim for the volume used for storing data for the Kafka broker pod `[idx]`.
+
+`data-[cluster-name]-zookeeper-[idx]`::
+Persistent Volume Claim for the volume used for storing data for the Zookeeper node pod `[idx]`.
+
+== Additional resources
+
+* For more information about ephemeral storage, visit xref:type-EphemeralStorage-reference[ephemeral storage schema reference].
```
Comment (dmuntima) [inline documentation/book/ref-storage.adoc]: Looks like the first two links are leading to this module. If yes, I don't think we need it here.

Comment (scholzj) [reply documentation/book/ref-storage.adoc]: No, they actually lead to the generated documentation, so I think we can leave them there.

**Pull request merged**